### DISCUSSION TITLE
How to set different timeouts for gPRC Exporters?

### DISCUSSION THREAD

**Discussion by angela278** [open-telemetry/opentelemetry-java#6260] [Q&A]
Hi all. We're using the OtlpGrpcSpanExporter, OtlpGrpcLogExporter, and OtlpGrpcMetricExporter in a sample application that hits our own OTel service. On the server side, we have an SLA of 60 seconds so we'd like to account for that on the client side by increasing the timeout to 60 seconds.

We tried using `.setTimeout(Duration.ofSeconds(60))` but saw that this only set the callTimeout in the internal OkHttpGrpcExporter - the connectTimeout, readTimeout, and writeTimeout are all still the default 10 seconds.

Is there a way for us to set the other timeouts as well? Or would writing our own exporters be a solution for this? TIA!

**Comment by jack-berg** [comment]
https://github.com/open-telemetry/opentelemetry-java/pull/6079 was just merged so you'll be able to set the connectTimeout once the next release is published. There still isn't any option for setting the read or write timeout. Can you describe your use case and why you believe those should be set? Thx.

**Reply by angela278** [reply]
Sure, we'd specifically like read timeout to be able to be set. For our use case, our server side SLA of 60 seconds can't be changed due to a dependency on Kafka and wait time for a Kafka response. Thus, we'd like our clients to have the option to set their read timeout to account for this rather than sending a RST_STREAM and retrying. 

Thank you for the response!
